### PR TITLE
store libass settings in subrenderersettings

### PR DESCRIPTION
**Pull request by adipose**

### CODE REVIEW

**Comment by adipose** [comment]
I ported this from an old branch I had mistakenly updated.  I think everything is working.

VSFilter settings I haven't done yet.  It's pretty trivial to add a reg setting for those if we want to enable libass from VSFilter.

In debug lite mode, I added back the link library for libass.  Previously I had made a way to exclude it from the build if libass wasn't in the props file, but it looks like you reverted that code.  But in lite mode, we need it to link (assuming libass enabled).  So I added it w/o the constant check.

File: src/Subtitles/STS.cpp [hunk 1]
```diff
@@ -2862,6 +2860,26 @@ bool CSimpleTextSubtitle::Open(CTextFile* f, int CharSet, CString name) {
...
+void CSimpleTextSubtitle::SetSubRenderSettings(SubRendererSettings settings) {
+    bool wasUsingLibass = subRendererSettings.renderUsingLibass;
+    subRendererSettings = settings;
+    if (settings.renderUsingLibass || wasUsingLibass) {
+        ResetASS();
+    }
+}
+
+void CSimpleTextSubtitle::ResetASS() {
+    UnloadASS();
```
Comment (clsid2) [inline src/Subtitles/STS.cpp]: For embedded subtitles a temporary file is created for parsing the header. When I was working on the code I had an issue where that triggered an unwanted call to UnloadASS(). I moved the call to inside LoadASSFile (after file existence check). So UnloadASS() should in this piece of code be moved to below "m_renderUsingLibass = false;"

Comment (adipose) [reply src/Subtitles/STS.cpp]: I added a fix for that, now.  I think it doesn't need to call it unless marked as loaded.

Previously I called that to clean up anything that might be left over.  Did it cause issues when it hadn't been previously loaded, or just a wasted call?

Comment (clsid2) [reply src/Subtitles/STS.cpp]: It was something like this:
Embedded sub -> data gets put in temp file for parsing -> sets m_path -> later call to SetDefaultStyle called UnloadASS

**Comment by adipose** [comment]
If it loads as a stream, there won't be a file. If the style is changed at that point it really should reload the stream. I think the extra loadassfile fixed issues with srt files, but maybe we need code to cache the stream too.

**Comment by clsid2** [comment]
I remember the cause now.
In your initial libass code in SubtitleInputPin.cpp the original open function was still called [pRTS->Open()]. I have added a check to only do that if m_assloaded is false.

**Comment by clsid2** [comment]
Can you add OpenTypeLang to the settings as well?

**Comment by clsid2** [comment]
I think we can optimize the code by calling stuff in this order:
DEBUG_NEW CRenderedTextSubtitle
SetSubRenderSettings
SetDefaultStyle
pRTS->Open

So apply settings before opening the subtitle file/stream, instead of after. I think that should avoid unnecessary reloads.

**Comment by clsid2** [comment]
And the libass reload should probably be moved from SetDefaultStyle to SetOverride?

**Comment by adipose** [comment]
> 
> 
> Can you add OpenTypeLang to the settings as well?

This is done.

**Comment by adipose** [comment]
> 
> 
> I think we can optimize the code by calling stuff in this order:
> DEBUG_NEW CRenderedTextSubtitle
> SetSubRenderSettings
> SetDefaultStyle
> pRTS->Open
> 
> So apply settings before opening the subtitle file/stream, instead of after. I think that should avoid unnecessary reloads.

Sounds good.  Some of the code will have to be reworked a bit since the "new" is inside of an if condition.

**Pull request merged**